### PR TITLE
fix #61(ranks): special ranks forum addon

### LINKED ISSUE #61: Add Forum Ranks Addon to Core
Hello, my proposal is to improve the forum from the ranks, the possibility of giving the rank, eg. Graphic, developer. In future of "webSPELL 5" make connect with the base in class

### PR DESCRIPTION
**Pull request by derchrisuk**
Adapted the Special Ranks Forum Addon to current dev branch.
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/webSPELL/webSPELL/pull/171%23issuecomment-70722189%22%2C%20%22https%3A//github.com/webSPELL/webSPELL/pull/171%23issuecomment-72085384%22%2C%20%22https%3A//github.com/webSPELL/webSPELL/pull/171%23issuecomment-72085434%22%2C%20%22https%3A//github.com/webSPELL/webSPELL/pull/171%23issuecomment-72180520%22%2C%20%22https%3A//github.com/webSPELL/webSPELL/pull/171%23discussion_r22924385%22%2C%20%22https%3A//github.com/webSPELL/webSPELL/pull/171%23discussion_r22945211%22%2C%20%22https%3A//github.com/webSPELL/webSPELL/pull/171%23discussion_r22966598%22%2C%20%22https%3A//github.com/webSPELL/webSPELL/pull/171%23discussion_r22966732%22%2C%20%22https%3A//github.com/webSPELL/webSPELL/pull/171%23discussion_r22966903%22%2C%20%22https%3A//github.com/webSPELL/webSPELL/pull/171%23discussion_r22967966%22%2C%20%22https%3A//github.com/webSPELL/webSPELL/pull/171%23discussion_r23792373%22%2C%20%22https%3A//github.com/webSPELL/webSPELL/pull/171%23discussion_r23792374%22%2C%20%22https%3A//github.com/webSPELL/webSPELL/pull/171%23discussion_r23793612%22%2C%20%22https%3A//github.com/webSPELL/webSPELL/commit/394f3d6705f98ba4439080e20561b2346da6f56a%23commitcomment-9504312%22%2C%20%22https%3A//github.com/webSPELL/webSPELL/commit/394f3d6705f98ba4439080e20561b2346da6f56a%23commitcomment-9504317%22%2C%20%22https%3A//github.com/webSPELL/webSPELL/commit/394f3d6705f98ba4439080e20561b2346da6f56a%23commitcomment-9504487%22%2C%20%22https%3A//github.com/webSPELL/webSPELL/commit/394f3d6705f98ba4439080e20561b2346da6f56a%23commitcomment-9504488%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/webSPELL/webSPELL/pull/171%23issuecomment-70722189%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Delete%20user%20ranks%20correctly%22%2C%20%22created_at%22%3A%20%222015-01-20T20%3A01%3A29Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1415280%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Phhere%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222015-01-29T19%3A14%3A37Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1415280%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Phhere%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Ashipit%3A%22%2C%20%22created_at%22%3A%20%222015-01-29T19%3A14%3A57Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/498197%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Pascalmh%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Ashipit%3A%22%2C%20%22created_at%22%3A%20%222015-01-30T10%3A18%3A07Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/706758%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/derchrisuk%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%206a10897bfb701fb13a1fbc508231c7c4797f4c52%20admin/ranks.php%2085%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/webSPELL/webSPELL/pull/171%23discussion_r22966903%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%2220%2B49%2B10%2B10%2B11%2B10%20%3D%20110%22%2C%20%22created_at%22%3A%20%222015-01-14T20%3A22%3A46Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/498197%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Pascalmh%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-complete%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-01-29T18%3A55%3A13Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/498197%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Pascalmh%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20admin/ranks.php%3AL188-195%22%7D%2C%20%22Pull%206a10897bfb701fb13a1fbc508231c7c4797f4c52%20admin/members.php%2025%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/webSPELL/webSPELL/pull/171%23discussion_r22966732%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Codestyle%3A%20add%20%5C%22%20.%20%24var%20.%20%5C%22%20for%20the%20whole%20query%22%2C%20%22created_at%22%3A%20%222015-01-14T20%3A20%3A14Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/498197%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Pascalmh%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222015-01-29T19%3A09%3A37Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/498197%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Pascalmh%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20admin/members.php%3AL166-181%22%7D%2C%20%22Pull%206a10897bfb701fb13a1fbc508231c7c4797f4c52%20templates/profile.html%205%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/webSPELL/webSPELL/pull/171%23discussion_r22967966%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Won%27t%20this%20add%20at%20unnecessary%20line%20breaks%20if%20user%20has%20not%20special%20rank%3F%22%2C%20%22created_at%22%3A%20%222015-01-14T20%3A37%3A46Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/498197%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Pascalmh%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-01-29T18%3A55%3A12Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/498197%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Pascalmh%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20templates/profile.html%3AL34-41%22%7D%2C%20%22Commit%20394f3d6705f98ba4439080e20561b2346da6f56a%20admin/ranks.php%2030%20167%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/webSPELL/webSPELL/commit/394f3d6705f98ba4439080e20561b2346da6f56a%23commitcomment-9504317%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Shouldn%27t%20it%20be%20%60%60%60%60id%3D%5C%22max%5C%22%60%60%60%60%22%2C%20%22created_at%22%3A%20%222015-01-29T18%3A44%3A48Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/706758%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/derchrisuk%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-01-29T18%3A55%3A35Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/498197%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Pascalmh%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20admin/ranks.php%3AL167%20%28394f3d6%29%22%7D%2C%20%22Pull%206a10897bfb701fb13a1fbc508231c7c4797f4c52%20admin/members.php%204%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/webSPELL/webSPELL/pull/171%23discussion_r22924385%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Can%27t%20we%20merge%20these%20Modules%3F%22%2C%20%22created_at%22%3A%20%222015-01-14T09%3A18%3A20Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/498197%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Pascalmh%22%7D%7D%2C%20%7B%22body%22%3A%20%22The%20language%20file%20is%20also%20added%20to%20other%20admin/.php%20files.%5Cr%5CnMerging%20would%20create%20duplicate%20entries.%22%2C%20%22created_at%22%3A%20%222015-01-14T15%3A50%3A52Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/706758%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/derchrisuk%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222015-01-14T20%3A18%3A04Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/498197%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Pascalmh%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20admin/members.php%3AL26-33%22%7D%2C%20%22Commit%20394f3d6705f98ba4439080e20561b2346da6f56a%20admin/ranks.php%2025%20163%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/webSPELL/webSPELL/commit/394f3d6705f98ba4439080e20561b2346da6f56a%23commitcomment-9504312%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Shouldn%27t%20it%20be%20%60%60%60%60id%3D%5C%22min%5C%22%60%60%60%60%22%2C%20%22created_at%22%3A%20%222015-01-29T18%3A44%3A16Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/706758%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/derchrisuk%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-complete%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-01-29T18%3A55%3A35Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/498197%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Pascalmh%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20admin/ranks.php%3AL163%20%28394f3d6%29%22%7D%7D%2C%20%22approved%22%3A%20%7B%22https%3A//github.com/Pascalmh%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/498197%3Fv%3D3%22%7D%2C%20%22https%3A//github.com/derchrisuk%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/706758%3Fv%3D3%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/Pascalmh'></a><a href='https://github.com/derchrisuk'></a>

- [x] <a href='#crh-comment-Pull 6a10897bfb701fb13a1fbc508231c7c4797f4c52 admin/members.php 4'></a> &nbsp;<b><a href='https://github.com/webSPELL/webSPELL/pull/171#discussion_r22924385'>File: admin/members.php:L26-33</a></b>
- <a href='https://github.com/Pascalmh'></a> Can't we merge these Modules?
- <a href='https://github.com/derchrisuk'></a> The language file is also added to other admin/.php files.
Merging would create duplicate entries.
- [x] <a href='#crh-comment-Pull 6a10897bfb701fb13a1fbc508231c7c4797f4c52 admin/members.php 25'></a> &nbsp;<b><a href='https://github.com/webSPELL/webSPELL/pull/171#discussion_r22966732'>File: admin/members.php:L166-181</a></b>
- <a href='https://github.com/Pascalmh'></a> Codestyle: add " . $var . " for the whole query
- [x] <a href='#crh-comment-Pull 6a10897bfb701fb13a1fbc508231c7c4797f4c52 admin/ranks.php 85'></a> &nbsp;<b><a href='https://github.com/webSPELL/webSPELL/pull/171#discussion_r22966903'>File: admin/ranks.php:L188-195</a></b>
- <a href='https://github.com/Pascalmh'></a> 20+49+10+10+11+10 = 110
- [x] <a href='#crh-comment-Pull 6a10897bfb701fb13a1fbc508231c7c4797f4c52 templates/profile.html 5'></a> &nbsp;<b><a href='https://github.com/webSPELL/webSPELL/pull/171#discussion_r22967966'>File: templates/profile.html:L34-41</a></b>
- <a href='https://github.com/Pascalmh'></a> Won't this add at unnecessary line breaks if user has not special rank?
- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/webSPELL/webSPELL/pull/171#issuecomment-70722189'>General Comment</a></b>
- <a href='https://github.com/Phhere'></a> Delete user ranks correctly
- [x] <a href='#crh-comment-Commit 394f3d6705f98ba4439080e20561b2346da6f56a admin/ranks.php 25 163'></a> &nbsp;<b><a href='https://github.com/webSPELL/webSPELL/commit/394f3d6705f98ba4439080e20561b2346da6f56a#commitcomment-9504312'>File: admin/ranks.php:L163 (394f3d6)</a></b>
- <a href='https://github.com/derchrisuk'></a> Shouldn't it be ````id="min"````
- [x] <a href='#crh-comment-Commit 394f3d6705f98ba4439080e20561b2346da6f56a admin/ranks.php 30 167'></a> &nbsp;<b><a href='https://github.com/webSPELL/webSPELL/commit/394f3d6705f98ba4439080e20561b2346da6f56a#commitcomment-9504317'>File: admin/ranks.php:L167 (394f3d6)</a></b>
- <a href='https://github.com/derchrisuk'></a> Shouldn't it be ````id="max"````

<a href='https://www.codereviewhub.com/webSPELL/webSPELL/pull/171?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/webSPELL/webSPELL/pull/171?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/webSPELL/webSPELL/pull/171?approve=0'></a>&nbsp;<a href='https://github.com/webSPELL/webSPELL/pull/171'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by derchrisuk** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/webSPELL/webSPELL-4.2.3/pull/171'>Refresh</a>.<a href='#crh-update-none'></a>

File: admin/members.php [hunk 1]
```diff
@@ -26,6 +26,7 @@
 */
 
 $_language->readModule('members');
+$_language->readModule('rank_special', true);
```
Comment (Pascalmh) [inline admin/members.php]: Can't we merge these Modules?

Comment (derchrisuk) [inline admin/members.php]: The language file is also added to other admin/.php files.
Merging would create duplicate entries.

Comment (Pascalmh) [inline admin/members.php]: :+1:

File: admin/members.php [hunk 2]
```diff
@@ -164,7 +166,15 @@
                 }
             }
 
-            safe_query("UPDATE " . PREFIX . "user SET userdescription='$userdescription' WHERE userID='$id'");
+            safe_query(
+                "UPDATE "
+                . PREFIX . "user
+                SET
+                    userdescription='$userdescription',
```
Comment (Pascalmh) [inline admin/members.php]: Codestyle: add " . $var . " for the whole query

File: admin/ranks.php [hunk 3]
```diff
@@ -166,6 +188,7 @@
     <tr>
       <td width="20%" class="title"><b>' . $_language->module[ 'rank_icon' ] . '</b></td>
       <td width="49%" class="title"><b>' . $_language->module[ 'rank_name' ] . '</b></td>
+      <td width="10%" class="title"><b>' . $_language->module[ 'special_rank' ] . '</b></td>
```
Comment (Pascalmh) [inline admin/ranks.php]: 20+49+10+10+11+10 = 110

File: templates/profile.html [hunk 4]
```diff
@@ -34,7 +34,7 @@
             </tr>
             <tr>
                 <th>%usertitle%:</th>
-                <td>$usertype<br>$rang</td>
+                <td>$usertype<br>$rang<br>$specialrang<br>$specialtype</td>
```
Comment (Pascalmh) [inline templates/profile.html]: Won't this add at unnecessary line breaks if user has not special rank?

**Comment by Phhere** [comment]
Delete user ranks correctly

File: templates/profile.html (hunk 4 above)
Comment (Pascalmh) [inline templates/profile.html]: :+1:<a href='#crh-update-none'></a>

File: admin/ranks.php (hunk 3 above)
Comment (Pascalmh) [inline admin/ranks.php]: :+1:<a href='#crh-update-complete'></a>

File: admin/members.php (hunk 2 above)
Comment (Pascalmh) [inline admin/members.php]: :+1:

**Comment by Phhere** [comment]
:+1:

**Comment by Pascalmh** [comment]
:shipit:

**Comment by derchrisuk** [comment]
:shipit:

**Pull request merged**